src/qililab/instruments/qblox/qblox_qrm.py

```diff
@@ -147,6 +147,17 @@ def compile(
         sequencers = self.get_sequencers_from_chip_port_id(chip_port_id=pulse_bus_schedule.port)
         # Separate the readout schedules by the qubit they act on
         qubit_schedules = pulse_bus_schedule.qubit_schedules()
+
+        # empty cache and sequence for specific qubits if they are not in the schedule
+        unused_seq = []
+        for cached_qubit, seq_id in (
+            (timeline.qubit, key) for key, element in self._cache.items() for timeline in element.timeline
+        ):
+            if cached_qubit not in (schedule.qubit for schedule in qubit_schedules):
+                unused_seq.append(seq_id)
+                _ = self.sequences.pop(seq_id)
+        self._cache = {key: items for key, items in self._cache.items() if key not in unused_seq}
```

Comment:
Couldn't we do this?
```suggestion
        for cached_qubit, seq_id in (
            (timeline.qubit, key) for key, element in self._cache.items() for timeline in element.timeline
        ):
            if cached_qubit not in (schedule.qubit for schedule in qubit_schedules):
                unused_seq.append(seq_id)
                _ = self.sequences.pop(seq_id)
                _ = self._cache.pop(seq_id)
```

Reply:
yup